### PR TITLE
[Andrei Tulpan] adaptors_task

### PR DESCRIPTION
**Pull request by andreitulpan**

### CODE REVIEW

File: adaptors_task/student/include/impl.h [hunk 1]
```diff
@@ -11,4 +12,38 @@
  *
  * @return true if expression is valid
  */
-bool isValid(const std::string& source);
+bool isValid(const std::string& source) {
+    std::stack<char> stack;
+    for (auto const &c : source) {
+        if (c == '(' || c == '[' || c == '{') {
```
Comment (smay1613) [inline adaptors_task/student/include/impl.h]: It's better to pack this into some container

File: adaptors_task/student/include/impl.h [hunk 2]
```diff
@@ -11,4 +12,38 @@
...
+            stack.push(c);
+        } else if (c == ')' || c == ']' || c == '}') {
+            if (stack.empty()) {
+                return false;
+            }
+            char ch = stack.top();
+            stack.pop();
+            switch (ch) {
```
Comment (smay1613) [inline adaptors_task/student/include/impl.h]: this should be definitely a search in the container

File: adaptors_task/student/include/impl.h [hunk 3]
```diff
@@ -11,4 +12,38 @@
...
+                    break;
+                case '{':
+                    if (c != '}') {
+                        return false;
+                    }
+                    break;
+            }
+        }
+    }
+    if (!stack.empty()) {
```
Comment (smay1613) [inline adaptors_task/student/include/impl.h]: please avoid naked true and false, just return stack.empty()

File: adaptors_task/student/include/impl.h [hunk 4]
```diff
@@ -11,4 +12,38 @@
  *
  * @return true if expression is valid
  */
-bool isValid(const std::string& source);
+bool isValid(const std::string& source) {
+    std::stack<char> stack;
```
Comment (smay1613) [inline adaptors_task/student/include/impl.h]: Please express the intent, what is the "stack"? Maybe openedBrackets?

Comment (andreitulpan) [reply adaptors_task/student/include/impl.h]: Yes, I wasn't to suggestive with this name. I resolved that.

File: adaptors_task/student/include/impl.h (hunk 3 above)
Comment (andreitulpan) [reply adaptors_task/student/include/impl.h]: done

File: adaptors_task/student/include/impl.h (hunk 1 above)
Comment (andreitulpan) [reply adaptors_task/student/include/impl.h]: I did it and the code is smaller now and can be extended much more simple.

File: adaptors_task/student/include/impl.h (hunk 2 above)
Comment (andreitulpan) [reply adaptors_task/student/include/impl.h]: done

File: adaptors_task/student/src/impl.cpp [hunk 5]
```diff
@@ -6,3 +6,28 @@ const std::vector<std::pair<char, char>> brackets {
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source) {
+    std::stack<char> openedBrackets;
+    for (auto const &c : source) {
+        for (auto const &currentBracket : brackets) {
+            if (c == currentBracket.first) {
```
Comment (smay1613) [inline adaptors_task/student/src/impl.cpp]: be more explicit like:
if (const bool isOpenBracket = (c == currentBracket.first))

**Pull request closed without merging**